src/main/kotlin/org/rust/ide/annotator/fixes/ChangeReturnTypeFix.kt

```diff
@@ -84,6 +84,11 @@ class ChangeReturnTypeFix(element: RsElement, private val actualTy: Ty) : LocalQ
             val isOverriddenFn = owner is RsFunction && owner.superItem != null
             if (isOverriddenFn) return null // TODO: Support overridden items
 
+            // do not offer in lambdas passed to functions
```

Comment:
It is still available in the following cases:

![image](https://user-images.githubusercontent.com/6079006/90123781-0943d880-dd68-11ea-9a45-e4779c43a424.png)

![image](https://user-images.githubusercontent.com/6079006/90123968-4e680a80-dd68-11ea-8488-b53dd8156b72.png)

I'm not sure how to fix this properly. Maybe we shouldn't offer the fix if we're inside a lambda and it doesn't have a return value?

Reply:
That seems like a better solution. If a lambda does not have an explicit return type, the return type has to be inferred from the environment, otherwise the plugin wouldn't show the error. In such case it indeed does not make sense to offer the fix, because the return type is "fixed" by the inference.

If the lambda has a return type specified, but it's wrong with respect to the surrounding context, it still won't compile after the fix is performed, but it will fix consistency of the lambda itself.